src/testers.jl

```diff
@@ -100,17 +100,20 @@ function test_frule(
     isapprox_kwargs = (; rtol=rtol, atol=atol, kwargs...)
 
     @testset "test_frule: $f on $(_string_typeof(args))" begin
-        xẋs = auto_primal_and_tangent.(args)
-        xs = primal.(xẋs)
-        tangents = (rand_tangent(f), tangent.(xẋs)...)
-        if check_inferred && _is_inferrable(f, deepcopy(xs)...; deepcopy(fkwargs)...)
-            _test_inferred(frule, (deepcopy(tangents)...,), f, deepcopy(xs)...; deepcopy(fkwargs)...)
+
+        primals_and_tangents = auto_primal_and_tangent.((f, args...))
+        func, xs = Iterators.peel(primal.(primals_and_tangents))
```

Comment:
It seems like most places (except line 116) that we use this, we could be using 
```suggestion
        primals = primal.(primals_and_tangents)
```
and for line 116 we might want to define a helper closure locally
`call_on_copy(f, xs...) = deepcopy(f)(deepcopy(xs)...; deepcopy(fkwargs)...)`
Which we also need for line 132.